### PR TITLE
Fix registry not loaded while retrieving consent manager receipt

### PR DESCRIPTION
**Pull request by tharindu-bandara**
### Proposed changes in this pull request

Initializing registry before retrieving consent manager receipt.

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java [hunk 1]
```diff
@@ -402,7 +404,7 @@ private Receipt getConsentReceiptOfUser(ServiceProvider serviceProvider, Authent
             }
         } catch (ConsentManagementException e) {
             throw new SSOConsentServiceException("Consent Management Error",
-                                                 "Error while retrieving user consents.", e);
+                    "Error while retrieving user consents.", e);
```
Comment (Abilashini) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java]: Shall we format the line by moving the 2nd parameter to the above line? 
```
            throw new SSOConsentServiceException("Consent Management Error", "Error while retrieving user consents.", 
                    e);
```

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java [hunk 2]
```diff
@@ -417,7 +419,7 @@ private AddReceiptResponse addReceipt(String subject, String subjectTenantDomain
             receiptResponse = getConsentManager().addConsent(receiptInput);
         } catch (ConsentManagementException e) {
             throw new SSOConsentServiceException("Consent receipt error", "Error while adding the consent " +
-                                                                                 "receipt", e);
+                    "receipt", e);
```
Comment (Abilashini) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java]: Shall we move this line to the previous line?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java [hunk 3]
```diff
@@ -544,7 +546,7 @@ private PIICategory addPIICategoryForClaim(ClaimMetaData claim) throws SSOConsen
             piiCategory = getConsentManager().addPIICategory(piiCategoryInput);
         } catch (ConsentManagementException e) {
             throw new SSOConsentServiceException("Consent PII category error", "Error while adding" +
-                                                                      " PII category:" + DEFAULT_PURPOSE_CATEGORY, e);
+                    " PII category:" + DEFAULT_PURPOSE_CATEGORY, e);
```
Comment (Abilashini) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java]: " PII category:" can be moved to the previous line. 
To improve the readability would be better to add a space after ":"

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java [hunk 4]
```diff
@@ -624,7 +626,7 @@ private Purpose addDefaultPurpose() throws SSOConsentServiceException {
             purpose = getConsentManager().addPurpose(defaultPurpose);
         } catch (ConsentManagementException e) {
             throw new SSOConsentServiceException("Consent purpose error",
-                                                 "Error while adding purpose: " + DEFAULT_PURPOSE, e);
+                    "Error while adding purpose: " + DEFAULT_PURPOSE, e);
```
Comment (Abilashini) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java]: `"Error while adding purpose: " + DEFAULT_PURPOSE,` can be moved to the previous line

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java [hunk 5]
```diff
@@ -987,17 +988,29 @@ private Receipt getReceipt(AuthenticatedUser authenticatedUser, String receiptId
...
+            initializeTenantRegistry(authenticatedUser);
             startTenantFlowWithUser(subject, authenticatedUser.getTenantDomain());
             currentReceipt = getConsentManager().getReceipt(receiptId);
         } catch (ConsentManagementException e) {
             throw new SSOConsentServiceException("Consent Management Error",
-                                                 "Error while retrieving user consents.", e);
+                    "Error while retrieving user consents.", e);
+        } catch (IdentityException e) {
+            throw new SSOConsentServiceException("Consent Management Error",
+                    "Error while initializing registry for the tenant domain: " + authenticatedUser.getTenantDomain(), e);
```
Comment (Abilashini) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java]: This line has been exceeded 120 characters

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java (hunk 1 above)
Comment (tharindu-bandara) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java]: Fixed in 06271985d0d22192422630521503f04db25e5eff.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java (hunk 2 above)
Comment (tharindu-bandara) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java]: Fixed in 06271985d0d22192422630521503f04db25e5eff.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java (hunk 3 above)
Comment (tharindu-bandara) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java]: Fixed in 06271985d0d22192422630521503f04db25e5eff.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java (hunk 4 above)
Comment (tharindu-bandara) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java]: Fixed in 06271985d0d22192422630521503f04db25e5eff.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java (hunk 5 above)
Comment (tharindu-bandara) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java]: Fixed in 06271985d0d22192422630521503f04db25e5eff.

**Pull request merged**